src/Controls/src/Xaml/ApplyPropertiesVisitor.cs

```diff
@@ -488,7 +488,29 @@ static bool TryConnectEvent(object element, string localName, bool attached, obj
 			var rootElementType = rootElement.GetType();
 			do
 			{
-				var mi = rootElementType.GetMethod((string)value, BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic);
+				MethodInfo mi = null;
+				var methodinfos = rootElementType.GetMethods(BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic)
+					.Where(mi => mi.Name == (string)value && mi.GetParameters().Length==2)
+					.ToArray();
+
+				if (methodinfos != null && methodinfos.Length == 1)
+					mi = methodinfos[0];
+				else
+				{
+					//try to find an event handler with a signature matching the button delegate signature
+					foreach (var methodinfo in methodinfos)
```

Comment:
Are there any other types that might also have multiple event handlers? Should we check for the Button type specifically here, or will this be agnostic?

Reply:
event _handlers_ are not in Button, but in user specific code. and this fixes apply to all event handlers, including ones defined on user controls